crates/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -0,0 +1,190 @@
+                            .unwrap_or_default()
+                    })
+                    .filter(|instance_key| !instance_key.is_splat())
+                    .map(|instance_key| InstancePath::instance(entity.clone(), instance_key))
+            })
+            .unique()
+            .collect();
+
+        let all_components: Vec<_> = entities
+            .iter()
+            .flat_map(|entity| {
+                store
+                    .all_components(&query.timeline, entity)
+                    .unwrap_or_default()
+            })
+            .unique()
+            .filter(|comp| !comp.is_indicator_component())
+            .collect();
+
+        egui::ScrollArea::both()
+            .auto_shrink([false, false])
+            .show(ui, |ui| {
+                egui::Frame {
+                    inner_margin: egui::Margin::same(5.0),
+                    ..Default::default()
+                }
+                .show(ui, |ui| {
+                    egui_extras::TableBuilder::new(ui)
+                        .columns(
+                            Column::auto_with_initial_suggestion(200.0),
+                            all_components.len() + 1,
+                        )
+                        .resizable(true)
+                        .vscroll(false)
+                        .auto_shrink([false, true])
+                        .striped(true)
+                        .header(re_ui::ReUi::table_line_height(), |mut row| {
+                            row.col(|ui| {
+                                ui.strong("Entity");
+                            });
+
+                            for comp in &all_components {
+                                row.col(|ui| {
+                                    ui.strong(comp.short_name());
+                                });
+                            }
+                        })
+                        .body(|body| {
+                            body.rows(
+                                re_ui::ReUi::table_line_height(),
+                                all_instances.len(),
+                                |idx, mut row| {
+                                    let instance = &all_instances[idx];
+
+                                    row.col(|ui| {
+                                        ui.label(format!("{instance}"));
+                                    });
+
+                                    for comp in &all_components {
+                                        row.col(|ui| {
+                                            if let Some((_, comp_inst)) =
+                                                get_component_with_instances(
+                                                    store,
+                                                    &latest_at_query,
+                                                    &instance.entity_path,
+                                                    *comp,
+                                                )
```

Comment:
As per our discussion yesterday, this is acceptable because the inner table closure is called _only_ for the displayed rows. I added a comment to that effect.